SU2_CFD/src/interfaces/cht/CConjugateHeatInterface.cpp

```diff
@@ -104,95 +93,112 @@ void CConjugateHeatInterface::GetDonor_Variable(CSolver *donor_solution, CGeomet
 
-    heat_flux_density       = thermal_conductivity*dTdn;
-    conductivity_over_dist  = thermal_conductivity/dist;
+      thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref();
+      conductivity_over_dist  = thermal_conductivity/dist;
+    }
   }
   else if (incompressible_flow) {
 
-    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
-
     thermal_conductivityND  = donor_solution->node[iPoint]->GetThermalConductivity();
-    thermal_conductivity = thermal_conductivityND*donor_config->GetConductivity_Ref();
+    heat_flux_density       = thermal_conductivityND*dTdn;
 
-    switch (donor_config->GetKind_ConductivityModel()) {
+    if (donor_config->GetCHT_Robin()) {
 
-      case CONSTANT_CONDUCTIVITY:
-        thermal_conductivity = thermal_conductivityND*donor_config->GetConductivity_Ref();
-        break;
+      switch (donor_config->GetKind_ConductivityModel()) {
 
-      case CONSTANT_PRANDTL:
-        thermal_conductivity = thermal_conductivityND*donor_config->GetGas_Constant_Ref()
-                               *donor_config->GetViscosity_Ref();
-        break;
-    }
+        case CONSTANT_CONDUCTIVITY:
+          thermal_conductivity = thermal_conductivityND*donor_config->GetConductivity_Ref();
+          break;
 
-    heat_flux_density       = thermal_conductivity*dTdn;
-    conductivity_over_dist  = thermal_conductivity/dist;
+        case CONSTANT_PRANDTL:
+          thermal_conductivity = thermal_conductivityND*donor_config->GetGas_Constant_Ref()
+                                 *donor_config->GetViscosity_Ref();
+          break;
+      }
+
+      conductivity_over_dist  = thermal_conductivity/dist;
+    }
   }
-  else if (flow) {
+  else if (heat_equation) {
 
-    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
+    /*--- Heat solver stand-alone case ---*/
 
-    thermal_conductivityND  = laminar_viscosity/Prandtl_Lam;
-    thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref()*cp_fluid;
+    thermal_diffusivity     = donor_config->GetThermalDiffusivity_Solid();
+    heat_flux_density       = thermal_diffusivity*dTdn;
 
-    heat_flux_density       = thermal_conductivity*dTdn;
-    conductivity_over_dist  = thermal_conductivity/dist;
+    if (donor_config->GetCHT_Robin()) {
+      rho_cp_solid            = donor_config->GetSpecific_Heat_Cp_Solid()*donor_config->GetDensity_Solid();
+      conductivity_over_dist  = thermal_diffusivity*rho_cp_solid/dist;
+    }
   }
-  else {
 
-    thermal_diffusivity     = donor_config->GetThermalDiffusivity_Solid();
-    heat_flux_density       = (thermal_diffusivity*dTdn)*rho_cp_solid;
-    conductivity_over_dist  = thermal_diffusivity*rho_cp_solid/dist;
+  /*--- Set the conjugate heat variables that are transferred by default ---*/
+
+  Donor_Variable[0] = Twall*donor_config->GetTemperature_Ref();
+  Donor_Variable[1] = heat_flux_density*donor_config->GetHeat_Flux_Ref();
```

Comment:
@economon Could you look over this and the `Heat_Fluxes` routine in the heat solver to check whether these usages of `GetHeat_Flux_Ref` meet your idea?

Reply:
See previous comment above..